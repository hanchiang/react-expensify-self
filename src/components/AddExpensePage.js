import React from 'react';

function AddExpensePage() {
  return (
    <div>Add expense page</div>
  );
}

export default AddExpensePage;