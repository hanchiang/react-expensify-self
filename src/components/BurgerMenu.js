import React from 'react';
import { slide as Menu } from 'react-burger-menu';
import { Link } from 'react-router-dom';

class BurgerMenu extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isOpen: false
    };
  }

  isMenuOpen(state) {
    return state.isOpen;
  }

  render() {
    return (
      <Menu right isOpen={this.state.isOpen} onStateChange={this.isMenuOpen}>
          <Link to="/profile">Profile</Link>
      </Menu>
    );
  }
}

export default BurgerMenu;