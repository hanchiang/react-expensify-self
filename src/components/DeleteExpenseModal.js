import React from 'react';
import Modal from 'react-modal';

Modal.setAppElement('#app');

function DeleteExpenseModal(props) {
  const handleConfirmDelete = (event) => props.onModalConfirm(true);
  const handleRejectDelete = (event) => props.onModalConfirm(false);

  return (
    <Modal
      isOpen={props.showModal}
      onRequestClose={props.onModalClose}
      closeTimeoutMS={300}
      contentLabel="Delete expense?"
      className="modal"
    >
      <div className="modal-header">
        <h2 className="modal-title">Delete Expense</h2>
      </div>

      <div className="modal-body">
        <p className="modal-message">Are you sure you want to delete expense?</p>
      </div>

      <div className="modal-actions">
        <button className="button button-normal" onClick={handleRejectDelete} type="button">Close</button>
        <button className="button button-warning" onClick={handleConfirmDelete} type="button">Delete</button>  
      </div>
    </Modal>
  );

}

export default DeleteExpenseModal;