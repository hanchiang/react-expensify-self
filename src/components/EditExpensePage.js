import React from 'react';

function EditExpensePage() {
  return (
    <div>Edit expense page</div>
  );
}

export default EditExpensePage;