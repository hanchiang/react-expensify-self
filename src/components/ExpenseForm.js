import React from 'react';
import { SingleDatePicker } from 'react-dates';
import moment from 'moment';

class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);
    
    const { expense } = this.props;
    this.state = {
      error: '',
      description: expense ? expense.description : '',
      amount: expense ? expense.amount/100 : '',
      note: expense ? expense.note : '',
      createdAt: expense ? moment(expense.createdAt) : moment(),
      focused: false
    };
  }

  onDateChange = (createdAt) => this.setState({createdAt});
  onFocusChange = ({focused}) => this.setState({focused});
  onDescriptionChange = (event) => this.setState({description: event.target.value});
  onAmountChange = (event) => {
    const amount = event.target.value;
    const amountRegex = /^\d+(\.\d{0,2})?$/;
    if (!amount || amount.match(amountRegex)) {
      this.setState({amount});
    }
  }
  onNoteChange = (event) => this.setState({ note: event.target.value });

  onSubmit = (event) => {
    event.preventDefault();
    let error = '';

    if (!this.state.description || !this.state.amount) {
      error = 'Please enter description and amount';
      this.setState({error});
    } else {
      this.setState({error});
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount) * 100,
        note: this.state.note,
        createdAt: this.state.createdAt.valueOf()
      });
    }
  }

  render() {
    const { error } = this.state;
    return (
        <form onSubmit={this.onSubmit} className="form">
          {error && <div className="form-error">{error}</div>}

          <input className="text-input" type="text" value={this.state.description} onChange={this.onDescriptionChange} placeholder="description" autoFocus />
          <input className="text-input" type="text" value={this.state.amount} onChange={this.onAmountChange} placeholder="Amount" />
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.focused}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
          />
          <textarea className="textarea" value={this.state.note} onChange={this.onNoteChange} placeholder="Add a note for your expense (optional)"></textarea>

          <div>
            <button className="button">Add Expense</button>
          </div>
        </form>
    );
  }
}

export default ExpenseForm;