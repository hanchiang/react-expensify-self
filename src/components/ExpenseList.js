import React from 'react';
import { connect } from 'react-redux';

import ExpenseListItem from './ExpenseListItem';
import getVisibleExpenses from '../selectors/getVisibleExpenses';

function ExpenseList(props) {
  const { expenses } = props;

  return (
    <div>
      {
        expenses.map(expense =>
          <ExpenseListItem key={expense.id} expense={expense} />
        )
      }
    </div>
  )
}

const mapStateToProps = (state) => ({
  expenses: getVisibleExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpenseList);