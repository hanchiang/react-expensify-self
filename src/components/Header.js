import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import BurgerMenu from './BurgerMenu';

export function Header(props) {
  const { displayName, photoURL } = props;
  return (
    <header className="header">
      <div className="content-container">
        <div className="header-content">
          <Link className="header-title" to="/dashboard">
            <h1>Expensify</h1>
          </Link>

          <div className="header-secondary">
            <div className="header-greetings">
              {photoURL && <img className="profile-pic show-for-desktop" src={photoURL} alt="Profile picture" height="42" />}
              {displayName && <span className="show-for-desktop greeting-text">Hello {displayName}!</span>}
            </div>         
          </div>

        </div>
      </div>

      <BurgerMenu />
    </header>
  );
}


const mapStateToProps = (state) => {
  const signInProvider = state.auth.signInProvider;
  const providerDataLength = state.auth.user && state.auth.user.providerData.length;
  
  return providerDataLength > 0 ? {
    displayName: state.auth.user.providerData[providerDataLength - 1].displayName,
    photoURL: state.auth.user.providerData[providerDataLength - 1].photoURL
  } :
  {displayName: '', photoURL: ''}
};


export default connect(mapStateToProps)(Header);