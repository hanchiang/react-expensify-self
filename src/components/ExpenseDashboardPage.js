import React from 'react';
import { NavLink } from 'react-router-dom';

import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

function ExpenseDashboardPage() {
    return (
        <div>
            <h1>Expense dashboard page</h1>
            <NavLink to="/create">Add expense</NavLink>

            <ExpenseListFilters />
            <ExpenseList />
        </div>
    )
}

export default ExpenseDashboardPage;