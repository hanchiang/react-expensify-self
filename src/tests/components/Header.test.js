import React from 'react';
import { shallow } from 'enzyme';

import { Header } from '../../components/Header';

test('should render header correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();

    // expect(wrapper.find('h1').length).toBe(1);
    // expect(wrapper.find('h1').text()).toBe('Expensify');
});
/*
test('should call startLogout on button click', () => {
    const startLogout = jest.fn();
    const wrapper = shallow(<Header startLogout={startLogout} />);
    wrapper.find('button').simulate('click');
    expect(startLogout).toHaveBeenCalled();
});
*/