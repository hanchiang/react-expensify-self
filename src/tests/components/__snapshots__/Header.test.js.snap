// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          Boilerplate
        </h1>
      </Link>
      <button
        className="button button--link"
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;
